test/test_other.py

```diff
@@ -8463,9 +8463,21 @@ def test_binaryen_stack_ir(self, opts, disable_and_enable):
   def test_run_order(self):
     create_file('lib.js', r'''
 addToLibrary({
+  foo__deps: ['$addOnPreRun', '$addOnInit', '$addOnPostCtor', '$addOnPreMain', '$addOnExit', '$addOnPostRun'],
   foo__postset: () => {
-    addAtPostCtor("console.log(`addAtPostCtor`);");
+    addAtPreRun("console.log(`addAtPreRun`);");
     addAtInit("console.log(`addAtInit`);");
+    addAtPostCtor("console.log(`addAtPostCtor`);");
+    addAtPreMain("console.log(`addAtPreMain`);");
+    addAtExit("console.log(`addAtExit`);");
+    addAtPostRun("console.log(`addAtPostRun`);");
+    // Add all the runtime equivalents of the above events.
+    return 'addOnPreRun(() => console.log("addOnPreRun"));\n' +
+           'addOnInit(() => console.log("addOnInit"));\n' +
+           'addOnPostCtor(() => console.log("addOnPostCtor"));\n' +
+           'addOnPreMain(() => console.log("addOnPreMain"));\n' +
+           'addOnExit(() => console.log("addOnExit"));\n' +
+           'addOnPostRun(() => console.log("addOnPostRun"));\n';
```

Comment:
I wonder if we should put these lines in a pre.js file?    I think that would make match more closely the use case.

Reply:
Should we include preInit in this test, so that we can be sure of the relative order?

Maybe we don't need `test_preinit` in that case since it seems to do that same thing

Reply:
Can you move the into a pre.js file so we don't have a `__postset` that has side effects *and* returns a string at the same time? 

I think that would be more obvious what is going on and I think it would how most ppl use these functions.